### PR TITLE
Use a URL, not a URI, for FetchEntry

### PR DESCRIPTION
**Pull request by alexwlchan**
This matches the spec slightly better, and lets me add tests/handling code for percent encoding of whitespace in URLs.

### CODE REVIEW

File: common/src/main/scala/uk/ac/wellcome/platform/archive/common/bagit/parsers/FetchContents.scala [hunk 1]
```diff
@@ -63,10 +63,16 @@ object FetchContents {
   def write(entries: Seq[FetchEntry]): String =
     entries
       .map { e =>
-        s"${e.url} ${encodeLength(e.length)} ${encodeFilepath(e.filepath)}"
+        s"${encodeURL(e.url)} ${encodeLength(e.length)} ${encodeFilepath(e.filepath)}"
       }
       .mkString("\n")
 
+  private def encodeURL(url: URL): String =
+    url.toString.replaceAll(" ", "%20").replaceAll("\t", "%09")
```
Comment (jamesgorrie) [inline common/src/main/scala/uk/ac/wellcome/platform/archive/common/bagit/parsers/FetchContents.scala]: Is there not a generic "en / decode all URL characters" - or is there a reason we don't use it?

Comment (alexwlchan) [reply common/src/main/scala/uk/ac/wellcome/platform/archive/common/bagit/parsers/FetchContents.scala]: Those methods exist, but the spec is a bit unclear about whether you should encode all the characters or just some of them (although I don’t think it matters).

**Comment by alexwlchan** [comment]
Closing as URI is arguably more correct; it supports `s3://` style URLs and gives us the code to decode them.
